cmd/createdkg.go

```diff
@@ -149,29 +187,81 @@ func runCreateDKG(ctx context.Context, conf createDKGConfig) (err error) {
 		return err
 	}
 
+	var privKey *k1.PrivateKey
+	creator := cluster.Creator{}
+
+	// Populate creator field
+	if conf.Publish {
+		// TODO(diogo): Should we store this private key in the disk?
+		// Temporary creator address
+		privKey, err = k1.GeneratePrivateKey()
+		if err != nil {
+			return errors.Wrap(err, "generate private key")
+		}
+		creator = cluster.Creator{
+			Address: eth2util.PublicKeyToAddress(privKey.PubKey()),
+		}
+	}
+
 	var opts []func(*cluster.Definition)
 	opts = append(opts, cluster.WithDKGAlgorithm(conf.DKGAlgo))
 	def, err := cluster.NewDefinition(
 		conf.Name, conf.NumValidators, conf.Threshold,
 		conf.FeeRecipientAddrs, conf.WithdrawalAddrs,
-		forkVersion, cluster.Creator{}, operators, conf.DepositAmounts,
+		forkVersion, creator, operators, conf.DepositAmounts,
 		conf.ConsensusProtocol, conf.TargetGasLimit, conf.Compounding,
 		crand.Reader, opts...)
 	if err != nil {
 		return err
 	}
-
 	if err := def.VerifyHashes(); err != nil {
 		return err
 	}
 
+	// Generate creator signature after hashes have been populated
+	if conf.Publish {
+		def.Creator.ConfigSignature, err = cluster.SignClusterDefinitionHash(privKey, def)
+		if err != nil {
+			return errors.Wrap(err, "sign cluster definition")
+		}
+	}
+
 	eth1Cl := eth1wrap.NewDefaultEthClientRunner(conf.ExecutionEngineAddr)
 	go eth1Cl.Run(ctx)
 
-	if err := def.VerifySignatures(eth1Cl); err != nil {
+	if err := def.VerifySignatures(eth1Cl, conf.Publish); err != nil {
```

Comment:
Do we even need to call that if we have `conf.Publish`? Aren't we verifying the signature we have just created 10 lines above?